### DISCUSSION TITLE
Discussion: Integration of Promotion Module version 1.20.1

### DISCUSSION THREAD

**Discussion by dtduong30** [medusajs/medusa#6252] [Q&A]
Hey everyone,

I'd like to initiate a discussion regarding the potential integration of the **Promotion** module (v1.20.1) into our project. This module could significantly enhance our project's functionality by providing campaign and promotions discount likes _"Buy 1 get many"._ However, upon adding the v1.20.1 package, I noticed discrepancies between the codebase and the documented features. Specifically, the models only contain the `promotion` and `application_methods`, which differs from the expected functionality  listed in the latest release's feature set [v1.20.1](https://github.com/medusajs/medusa/releases/tag/v1.20.1)

**Current State**: The module's code, as observed upon installation, does not correspond to the documented features provided in the GitHub release.
Integration Plan: What steps can we take to integrate the module despite the observed discrepancies?
Documentation Updates: When can we expect the documentation to include step-by-step implementation guidance for the intended functionality?

**Discussion Points**:

1. How can we integrate the module despite the inconsistencies?"
2. Are there other ways or solutions we should explore?"
3. When can we expect updated documentation and code?"

**Comment by adrien2p** [comment]
The modules are in beta and not really usable like that, they are planed for v2, that is why depending on when you install them and their version there might be some models and functionalities missing as well as some missing end points since they are under development 👍

cc. @olivermrbl @srindom

**Reply by dtduong30** [reply]
Thank you for explaining the current status of the modules. Since the Promotion module is in beta, I'm considering creating a custom solution to integrate _promotions with campaign_ into our project.

Could you advise on how to approach this customization to avoid conflicts between the v1 and v2 versions of the module? Specifically, what issues might arise, and how can we address them?

Also, I'm interested in knowing when we can expect the stable release of v2 and if there will be any changes to the process of adding items to the cart. @adrien2p  @olivermrbl  @srindom 

I appreciate your help in navigating these matters. Thank you!

**Reply by alessioacella** [reply]
@dtduong30 have you created a custom solution to integrate promotions with campaign in your project? 

We have the same needs in our v1.x project!
